### PR TITLE
Made AI move for Run and Rumble if immobile

### LINKED ISSUE #889: AI cells should decide to move immediately
Currently the AI waits about 3 seconds after a cell spawns before it does anything. It would be better to have the cells immediately think and start moving. This is only really visible when respawning.

### PR DESCRIPTION
**Pull request by Maxonovien**
**Brief Description of What This PR Does**

This PR modifies run and rumble to make immobile microbe move randomly (as they don't have any input to compute gradient from).
This makes newly spawned microbes move immediately (as they would previously only have 50% chance to move per step).
Note that this does not affect sessile organisms.

**Related Issues**

Closes #889 .

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed
- [x] Functionality is confirmed working by another person (see above checklist link)
- [ ] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

**Comment by Maxonovien** [comment]
As a note: while initializing microbes to move in any direction makes sense to me, it comes with microbes sometimes heading towards the player and pushing it for a while. There are 3 possible solutions in my mind:
1. Tweaking the possible orientations (only works if orientation on spawn is not fully random; and I don't think it is?)
2. Making microbes avoid the player in general.
3. Simply not caring, the player can move out of the way if wanted.

File: src/microbe_stage/MicrobeAI.cs [hunk 1]
```diff
@@ -367,6 +367,14 @@ private void EngagePrey(Vector3 target, Random random, bool engulf)
     // For doing run and tumble
     private void RunAndTumble(Random random)
     {
+        // If this microbe is actually immobile, just initialize by moving in a random direction.
```
Comment (adQuid) [inline src/microbe_stage/MicrobeAI.cs:370]:
```suggestion
        // If this microbe is currently stationary, just initialize by moving in a random direction.
```

Being picky with English, but normally "immobile" means that something CAN'T move, as opposed to not currently moving.

File: src/microbe_stage/MicrobeAI.cs [hunk 2]
```diff
@@ -367,6 +367,14 @@ private void EngagePrey(Vector3 target, Random random, bool engulf)
...
+        // Used to get newly spawned microbes to move.
+        if (microbe.MovementDirection.Length() == 0)
+        {
+            MoveWithRandomTurn(0, 2 * Mathf.Pi, random);
```
Comment (adQuid) [inline src/microbe_stage/MicrobeAI.cs:374]:
```suggestion
            MoveWithRandomTurn(0, Mathf.Pi, random);
```

MoveWithRandomTurn picks a direction and turns relative to the current heading, so just pi should be enough to be able to turn any direction.

**Review by adQuid — CHANGES_REQUESTED** [review]
Currently unable to test this, but I see a couple things in the code.

File: src/microbe_stage/MicrobeAI.cs (hunk 1 above)
Comment (Maxonovien) [reply src/microbe_stage/MicrobeAI.cs]: If I may say, you really should consider leaving us a memo when you change the meaning of the words you borrow from us. :p Actually was actually intended, but both case works so I'll just leave it like this. Thanks!

Comment (adQuid) [reply src/microbe_stage/MicrobeAI.cs]: I'll get back to you with a list of all culture America has ruined, but that might be beyond the scope of this PR.

**Review by adQuid — APPROVED** [review]
Can't prove it works, but the code looks good.

**Comment by adQuid** [comment]
Ran playtesting. I'm not seeing any issues with the AI and it does improve behavior. I'd like to see this merged, but I'm not sure if we can say this closes #889 

That issue seems to be describing a delay before even calling the AI code instead of a delay in the AI itself. I've tried to find an example of this happening (I recall observing it), but I wasn't able to in my 15 minutes of testing. On the other hand, this issue is 2 years old, and may have just gone away some other way.

@hhyyrylainen thoughts?

**Comment by Maxonovien** [comment]
I thought similarly at first, but the code actually sets new microbeAI to think at the first think step (because their TimeUntilNextUpdate is set to 0, and only got reset at AI_THINK_INTERVAL upon doing the first move). While it was the first reason for the issue, it is fixed already. I think this is what hh pointed out in the last comment on the issue.

**Comment by hhyyrylainen** [comment]
@adQuid I debugged #889 quite a bit and determined that the AI update for a new cell is ran on the next frame. It's just the AI decides very often to not move. That's why the issue is worded like it is.

**Comment by adQuid** [comment]
In which case I've checked off this ticket as tested.

**Comment by hhyyrylainen** [comment]
Does this still allow really lazy species (with the personality values) to not move when spawned? If so I'm happy with this also.

**Comment by Maxonovien** [comment]
The current changes only affect Run & Rumble algorithm, which only triggers if `SpeciesActivity > Constants.MAX_SPECIES_ACTIVITY / 10`. So sessile organisms, which do not qualify, should not be affected.

No difference is made otherwise, but this is the case for the whole movement algorithm in general.

**Comment by adQuid** [comment]
Well update the branch and let's see if we got a merge on our hands!

**Review by hhyyrylainen — APPROVED** [review]

**Comment by buckly90** [comment]
Appears to work great without a hitch. Didn't notice anything unusual as I played, and maybe had a bit too much fun endlessly duplicating my cell to watch the cells quickly float away like balloons.

**Comment by adQuid** [comment]
I admit I really did like the cone effect from before this change. It's like I was blasting a firehose of other cells.

**Pull request merged**